invokeai/frontend/web/src/features/controlLayers/hooks/useExportCanvasToPSD.ts

```diff
@@ -151,7 +52,7 @@ export const useExportCanvasToPSD = () => {
         return;
       }
 
-      if (canvasWidth > 30000 || canvasHeight > 30000) {
+      if (visibleRect.width > 30000 || visibleRect.height > 30000) {
```

Comment:
Where does the 30000 number come from? Is this a constraint inherited from `ag-psd`? Arbitrary? Determined by experiment? It's entirely possible that in practice, we can only export canvases of a much smaller size. 30000x30000 is _huge_, potentially gigabytes of image data.

Real-world testing will provide the most reasonable max size, likely constrained by memory usage. 

Extract it to a constant and explain _why_ we have chosen this value.

Finally, consider using an _area_ constraint instead of width and height.

Reply:
Arbitrary, created a max area with a max_dim * max_dim for easier tuning here, nad started at 8k